plugins/languages/treesitter/treesitter.nix

```
@@ -6,210 +6,317 @@
+    highlight = {
+      additional_vim_regex_highlighting =
+        with helpers.nixvimTypes;
+        helpers.defaultNullOpts.mkNullableWithRaw (either bool (listOf (maybeRaw str))) false ''
+          Setting this to true will run `:h syntax` and tree-sitter at the same time. \
+          Set this to `true` if you depend on 'syntax' being enabled (like for indentation). \
+          Using this option may slow down your editor, and you may see some duplicate highlights. \
+          Instead of true it can also be a list of languages
         '';
-      };
 
-      ignoreInstall = mkOption {
-        type = types.listOf types.str;
-        default = [ ];
-        description = "List of parsers to ignore installing (for \"all\")";
-      };
+      enable = helpers.defaultNullOpts.mkBool false ''
+        Whether to enable treesitter highlighting.
+      '';
 
-      disabledLanguages = mkOption {
-        type = types.listOf types.str;
-        default = [ ];
-        description = "A list of languages to disable";
-      };
+      disable =
+        with types;
+        helpers.defaultNullOpts.mkStrLuaFnOr (listOf str) null ''
+          Can either be a list of the names of parsers you wish to disable or
+          a lua function that returns a boolean indicating the parser should be disabled.
+        '';
 
-      customCaptures = mkOption {
-        type = types.attrsOf types.str;
-        default = { };
-        description = "Custom capture group highlighting";
-      };
+      custom_captures = helpers.defaultNullOpts.mkAttrsOf types.str { } ''
+        Custom capture group highlighting.
+      '';
+    };
+
+    incremental_selection = {
+      enable = helpers.defaultNullOpts.mkBool false ''
+        Incremental selection based on the named nodes from the grammar.
+      '';
 
-      incrementalSelection =
+      keymaps =
         let
-          keymap =
-            default:
-            mkOption {
-              type = types.str;
-              inherit default;
-              description = "Key shortcut";
-            };
+          mkKeymap = default: helpers.defaultNullOpts.mkStr default "Key shortcut";
         in
         {
-          enable = mkEnableOption "incremental selection based on the named nodes from the grammar";
-          keymaps = {
-            initSelection = keymap "gnn";
-            nodeIncremental = keymap "grn";
-            scopeIncremental = keymap "grc";
-            nodeDecremental = keymap "grm";
-          };
+          init_selection = mkKeymap "gnn";
+          node_incremental = mkKeymap "grn";
+          scope_incremental = mkKeymap "grc";
+          node_decremental = mkKeymap "grm";
         };
+    };
 
-      indent = mkEnableOption "tree-sitter based indentation";
+    indent = {
+      enable = helpers.defaultNullOpts.mkBool false ''
+        Whether to enable tree-sitter based indentation.
```

```suggestion
        Whether to enable treesitter based indentation.
```

I think we don't have the hyphen elsewhere?

---

I think the hyphen is for the actual binary. https://tree-sitter.github.io/tree-sitter/

---

I'm find with whatever you think is correct